### ISSUE TITLE
Client Startup: services.AddRemoting defaults to ServerRemotingExtensions.AddRemoting, causing runtime exception

### ISSUE THREAD

**Issue by weebs** [fsbolero/Bolero#22]
I ran into this issue today while trying to add remoting to the Bolero client side project. It seems that when I call services.AddRemoting(), it calls one of the ServerRemotingExtensions methods, and produces the following runtime error

> System.ArgumentException: 'Cannot instantiate implementation type 'Bolero.Remoting.IRemoteHandler' for service type 'Bolero.Remoting.IRemoteHandler'.'

Sample code:

```fsharp
namespace RemoteTest.Client

open Microsoft.AspNetCore.Blazor.Builder
open Microsoft.AspNetCore.Blazor.Hosting
open Microsoft.Extensions.DependencyInjection
open Bolero.Remoting

type Startup() =

    member __.ConfigureServices(services: IServiceCollection) =
        // Causes a runtime error
        services.AddRemoting() |> ignore
        // Works as expected
        ClientRemotingExtensions.AddRemoting(services) |> ignore

    member __.Configure(app: IBlazorApplicationBuilder) =
        app.AddComponent<Main.MyApp>("#main")

module Program =

    [<EntryPoint>]
    let Main args =
        BlazorWebAssemblyHost.CreateDefaultBuilder()
            .UseBlazorStartup<Startup>()
            .Build()
            .Run()
        0
```

**Comment by Tarmil** [comment]
Hmm, that's annoying indeed. It's strange that it defaults to the version that has a type parameter when you don't give one. Looks like we might have to either rename the methods (eg `AddClientRemoting` / `AddServerRemoting`) or move one of the extension classes to a submodule (eg `open Bolero.Remoting.Server`).

**Comment by Tarmil** [comment]
This was fixed in v0.4, where both methods were moved to submodules (`Bolero.Remoting.Client` and `Bolero.Remoting.Server`, respectively).
